packages/ui/src/theme/createTheme/resolveObject.ts

```diff
@@ -0,0 +1,169 @@
+// Internal Style Dictionary methods
+// copied from amzn/style-dictionary with the owner's permission
+import {
+  cloneDeep,
+  createReferenceRegex,
+  getName,
+  getPathFromName,
+  GroupMessages,
+  has,
+  resolveReference,
+  usesReference,
+} from '../../utils';
+
+const PROPERTY_REFERENCE_WARNINGS =
+  GroupMessages.GROUP.PropertyReferenceWarnings;
+
+let current_context = []; // To maintain the context to be able to test for circular definitions
+const defaults = {
+  ignoreKeys: ['original'],
+  ignorePaths: [],
+};
+let updated_object, regex, options;
+
+export function resolveObject<T>(object: Record<string, any>): T {
```

Comment:
The only change is adding the generic type `<T>`. This is modeled after the way we copied deepExtend [here](https://github.com/aws-amplify/amplify-ui/pull/5848/files#diff-fb83d2402d91c982f9aa9895fb73367a4f87aeb1da8d04bb4754bbda24086906L204)